examples/guard_riverpod/lib/main.dart

```diff
@@ -0,0 +1,23 @@
+
+import 'package:beamer/beamer.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:guard_riverpod/router.dart';
+
+void main() => runApp(ProviderScope(child: MyApp()));
+
+class MyApp extends ConsumerWidget {
+  final _scaffoldKey = GlobalKey<ScaffoldMessengerState>();
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return MaterialApp.router(
+      debugShowCheckedModeBanner: false,
+      // Passing `ref.read`, which is a `Reader` instance, so that we can use
+      // riverpod as we see fit.
+      routerDelegate: createDelegate(ref.read),
```

Comment:
@matuella i am not sure if passing the `WidgetRef`'s `read` would work long term. The latest [riverpod docs](https://github.com/rrousselGit/river_pod/blob/master/website/docs/concepts/reading.mdx#obtaining-a-ref-from-a-widget) say:

> When a widget obtains a "ref", this "ref" should not be passed around. It should be used only by the widget that created this object. 

To stay within the these guidelines of the doc, how about moving the delegate into main (also see this [thread](https://github.com/rrousselGit/river_pod/issues/295#issuecomment-770368500) regarding this pattern below):

```dart
void main() {
  final container = ProviderContainer();
  final delegate = createDelegate(container.read); /// ref could also be pushed into delegate, according to the docs.
  runApp(UncontrolledProviderScope(container:container, child: MyApp(delegate: delegate));
}
```

Reply:
Sorry, you're completely correct, thanks a lot for this suggestion!